src/Wine.cc

```diff
@@ -0,0 +1,175 @@
+/* -*- Mode: C++; tab-width: 8; c-basic-offset: 2; indent-tabs-mode: nil; -*- */
+
+#include <algorithm>
+#include <cstdint>
+#include <list>
+#include <sstream>
+#include <string>
+
+#include "AddressSpace.h"
+#include "Task.h"
+#include "log.h"
+
+using namespace std;
+
+namespace rr {
+
+/* see wine include/winnt.h*/
+#define IMAGE_DOS_SIGNATURE    0x5A4D     /* MZ   */
+#define IMAGE_NT_SIGNATURE     0x00004550 /* PE00 */
+#define IMAGE_NT_OPTIONAL_HDR32_MAGIC      0x10b
+#define IMAGE_NT_OPTIONAL_HDR64_MAGIC      0x20b
+struct __attribute__((packed)) IMAGE_DOS_HEADER {
+  uint16_t e_magic;  /* 00: MZ Header signature */
+  char unused1[0x3a];
+  uint32_t e_lfanew; /* 3c: Offset to extended header */
+};
+struct __attribute__((packed)) IMAGE_FILE_HEADER {
+  char unused1[0x2];
+  uint16_t NumberOfSections;
+  char unused2[0xc];
+  uint16_t SizeOfOptionalHeader;
+  char unused3[0x2];
+};
+struct __attribute__((packed)) IMAGE_OPTIONAL_HEADER32 {
+  uint16_t Magic;
+  char unused1[0xde];
+};
+struct __attribute__((packed)) IMAGE_NT_HEADERS {
+  uint32_t Signature;
+  struct IMAGE_FILE_HEADER FileHeader;
+  struct IMAGE_OPTIONAL_HEADER32 OptionalHeader;
+};
+struct __attribute__((packed)) IMAGE_SECTION_HEADER {
+  char unused1[0xc];
+  uint32_t VirtualAddress;
+  char unused2[0x18];
+};
```

Comment:
What's the copyright status of this code?

Reply:
Lines 18, 19, 20, 21, 23, 25, 48, 49, 50 are copied and adapted from [wine's winnt.h](https://source.winehq.org/git/wine.git/blob/HEAD:/include/winnt.h), which is licensed as LGPL as far as I see.
The remaining of the structure definitions I wrote myself by looking at the above file and the output of `wine winedbg --gdb notepad` with `Wine-gdb> ptype /o IMAGE_DOS_HEADER`.
Is this a problem with RRs license?
Would [mingw headers](https://sourceforge.net/p/mingw-w64/mingw-w64/ci/master/tree/mingw-w64-headers/include/winnt.h#l6134) with ZPL license be possible?
Or [boost headers](https://github.com/boostorg/dll/blob/develop/include/boost/dll/detail/pe_info.hpp#L42)?
Is it sufficient to remove the defines and rewrite the other lines?

Reply:
We don't want LGPL code in rr. The Boost license would be fine, but importing that Boost file as-is is probably unnecessarily complicated.

I suggest just rewriting the code from scratch, yourself, referring to the Boost file as needed, and mention in comments that you did that and that the Boost file is under the Boost (MIT-ish) license.

Reply:
Microsoft has an MIT licensed version of these struct defines here: https://github.com/microsoft/xlang/blob/b950b15d53cab2dddc78c65e783ef43042a2e8c9/src/library/impl/meta_reader/pe.h